### ISSUE TITLE
Provide a `| highlight` filter as well?

### ISSUE THREAD

**Issue by mirisuzanne** [11ty/eleventy-plugin-syntaxhighlight#15]
The paired shortcode is great when I know that an entire block rendered in my template will be code - but sometimes I'm rendering arbitrary markdown that might have code blocks nested inside. In that case, I'd love to do something like:

```njk
{{ data | markdown | highlight | safe }}
```

For now, I'll try importing prism as a stand-alone in addition to `eleventy-plugin-syntaxhighlight` - and use it to generate a custom filter (or tag along on my markdown filter).

**Comment by schneyra** [comment]
I'd love this, too. I'm using a CMS as source for my blogposts and it delivers pre-rendered HTML where code is embedded. 

```html
<pre class="language-css"><code>[...]</code><pre>
```
I hope I haven't overlooked anything, but I'd like to do something like `{{ article.content | highlight | safe }}`.

**Comment by jinsupark** [comment]
Just ran into this problem! This would be super helpful indeed.

**Comment by zachleat** [comment]
You _can_ DIY this with something like:

```js
const { pairedShortcode } = require("@11ty/eleventy-plugin-syntaxhighlight");

module.exports = function(eleventyConfig) {
  eleventyConfig.addFilter("highlight", function(content, language) {
    return pairedShortcode(content, language);
  });
};
```

**Comment by mirisuzanne** [comment]
@zachleat I don't think that quite solves the initial issue here - which is that only some of the `content` might involve fenced code, and only that code should be highlighted. If I understand this right, it would turn the entire `content` into a code-block instead.

**Comment by stillingdesign** [comment]
Just wanted to check-in on this and see if anyone has found a good solution? Running into this same problem.

**Comment by Snapstromegon** [comment]
Even though I didn't take a closer look at this plugin, IMO it should be possible to do this.

I personally switched away to using shiki as a highlighter and wrote a short blogpost on how to do it:
https://www.hoeser.dev/blog/2023-02-07-eleventy-shiki-simple/

The code from the TLDR should be possible to adapt to also allow for a highlight filter too.

**Comment by zachleat** [comment]
Sorry for the super late reply here but you have a few options:

Using the render plugin for Markdown (the `renderContent` filter specifically) will highlight markdown code blocks using the Eleventy Syntax Highlighter plugin: https://www.11ty.dev/docs/plugins/render/#render-content-filter

Otherwise, you can wire this up like so:

```js
import { pairedShortcode } from "@11ty/eleventy-plugin-syntaxhighlight";
import MarkdownIt from "markdown-it";

export default function(eleventyConfig) {
  let md = new MarkdownIt();
  md.set({
    html: true,
    highlight: function(code, language) {
      return pairedShortcode(code, language);
    }
  });
  md.disable("code");

  eleventyConfig.setLibrary("md", md);
  eleventyConfig.addFilter("markdown", async function(content) {
    return md.render(content);
  });
};
```

Take note that the `setLibrary` line overrides the internal Markdown library in Eleventy (and is optional, I suppose!) but I’ve kept the same defaults here that we use internally https://www.11ty.dev/docs/languages/markdown/#default-options

Sample Liquid usage:

````
{% capture arbitrary %}
# Heading

```html
<html></html>
```
{% endcapture %}
{{ arbitrary | markdown }}
````
